presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -332,6 +333,16 @@ private void processNewSource(TaskSource source)
     {
         checkLockHeld("Lock must be held to call processInternal");
 
+        // revoke operators memory
+        // do this synchronously for now
+        for (int i = 0; i < operators.size() && !driverContext.isDone(); i++) {
+            Operator current = operators.get(i);
+            if (current.getOperatorContext().isSystemMemoryRevokingRequested()) {
+                getUnchecked(current.revokeMemory());
```

Comment:
This shouldn't be synchronous. It can block multiple drivers for very long period of time (for example for the duration of spilling multiple giga bytes of data).